extractor/src/main/java/com/example/extractor/MainLauncher.java

```diff
@@ -0,0 +1,78 @@
+        String outputDirectory = "generated/AbbreviationAsWordInName";
+
+        // Process files in the input directory and save results to the output directory
+        processFiles(inputDirectory, outputDirectory);
+    }
+
+    public static void processFiles(String inputDir, String outputDir) throws Exception {
+        // Pattern to match files named Example#.java
+        Pattern pattern = Pattern.compile("Example\\d+\\.java");
+
+        // Collect all Example#.java files in the input directory
+        System.out.println("Walking through the input directory to collect Example#.java files...");
+        try (Stream<Path> paths = Files.walk(Paths.get(inputDir))) {
+            List<String> exampleFiles = paths
+                    .filter(Files::isRegularFile)
+                    .filter(path -> {
+                        Matcher matcher = pattern.matcher(path.getFileName().toString());
+                        return matcher.matches();
+                    })
+                    .map(Path::toString)
+                    .collect(Collectors.toList());
+
+            System.out.println("Found " + exampleFiles.size() + " Example#.java files.");
+
+            // Determine the template file path relative to the project root
+            String templateFilePath = "extractor/src/main/resources/config-template-treewalker.xml";
+
+            // Ensure output directory exists
+            Path outputPath = Paths.get(outputDir).toAbsolutePath();
+            if (!Files.exists(outputPath)) {
+                System.out.println("Output directory does not exist. Creating: " + outputPath);
+                Files.createDirectories(outputPath);
+            } else {
+                System.out.println("Output directory already exists: " + outputPath);
+            }
+
+            // Process each file and generate corresponding config
+            for (String exampleFile : exampleFiles) {
+                System.out.println("Processing file: " + exampleFile);
+                String fileContent = new String(Files.readAllBytes(Paths.get(exampleFile)));
+                System.out.println("File content:\n" + fileContent);
+
+                String generatedContent;
+                try {
+                    System.out.println("Generating configuration for file: " + exampleFile);
+                    generatedContent = ConfigSerializer.serializeConfigToString(exampleFile, templateFilePath);
+                    System.out.println("Generated configuration:\n" + generatedContent);
+                } catch (Exception e) {
+                    System.err.println("Failed to process file: " + exampleFile);
+                    e.printStackTrace();
+                    continue;
+                }
+
+                String fileName = Paths.get(exampleFile).getFileName().toString().replace(".java", "-config.xml");
```

Comment:
you need to write in folder-file structure like https://github.com/checkstyle/test-configs/tree/main/AbbreviationAsWordInName